presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java

```diff
@@ -73,58 +73,63 @@ private boolean isSupportedType(Type type)
         return type instanceof VarcharType || SUPPORTED_PRIMITIVE_TYPES.contains(type);
     }
 
+    private String currentColumnMapping()
+    {
+        return columnHandles.get(currentColumnIndex).getMapping();
```

Comment:
This is "the fix". Rename `Add Kafka Avro write smoke tests`
to "Fix ...", eg "Fix Kafka Avro encoder to use mappings"

Reply:
One thing I'd like to highlight is that this commit is breaking pre-existing behaviour. Earlier people could've not provided a `mapping` and we would have INSERTed using the column name. Now `mapping` is no longer optional on the INSERT path.

Also keep in mind that the mapping can only be used for Avro and JSON, not RAW or CSV since the meaning of mapping differs across encoders (and connectors).